src/main/java/practice/StreamPractice.java

```diff
@@ -59,7 +86,14 @@ public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        Predicate<Person> predicate = person -> person.getAge() >= femaleAge
+                && person.getSex() == Person.Sex.WOMAN;
+        return peopleList.stream()
+                .filter(predicate)
+                .map(Person::getCats)
+                .flatMap(Collection::stream)
```

Comment:
```suggestion
                .flatMap(p -> p.getCats().stream())
```
as described in common mistakes you should use a minimum amount of stream operations